packages/prisma-client/prisma/migrations/20221218185551_normalized_tree/migration.ts

```diff
@@ -0,0 +1,73 @@
+import { PrismaClient } from "./client";
+
+type Instance = {
+  id: string;
+  component: any;
+  cssRules: Array<any>;
+  children: Array<Instance | string>;
+};
+
+type NormalizedInstance = {
+  id: string;
+  component: any;
+  cssRules: Array<any>;
+  children: (number | string)[];
+};
+
+const normalizeTree = (rootInstance: Instance) => {
+  const normalizedTree: NormalizedInstance[] = [];
+  const traverse = (
+    instance: Instance,
+    normalizedInstance: NormalizedInstance
+  ) => {
+    for (const child of instance.children) {
+      if (typeof child === "string") {
```

Comment:
should we also make child always an instance, no matter if its a string?

Reply:
We could distinct like this
```
| { type: 'text', value: '' }
| { type: 'instance', children: [] }
```

Reply:
yep

Reply:
Done https://github.com/webstudio-is/webstudio-designer/pull/654